### PR TITLE
DEX-1057 Write order to DB once

### PR DESCRIPTION
**Pull request by JennaWestenra**

### CODE REVIEW

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala [hunk 1]
```diff
@@ -50,6 +78,18 @@ class MatcherRestartTestSuite extends MatcherSuiteBase {
...
+        dex1.api.getOrderStatus(order).status shouldBe HttpOrderStatus.Status.Accepted
+      }
+
+      dex1.api.getOrderStatus(bobsOrderToFulfil).status shouldBe HttpOrderStatus.Status.Filled
+      dex1.api.getOrderStatus(bobsOrderToCancel).status shouldBe HttpOrderStatus.Status.Cancelled
+
     }
   }
+
+  private def mkBobOrder() = mkOrder(bob, wctWavesPair, OrderType.SELL, Gen.choose(10, 500).sample.get, 2.wct * Order.PriceConstant)
```
Comment (vsuharnikov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:93]: There is a chance, that Gen selects two equal values. In such cases we introduce a parameter `i` and:
* either increase an amount or a price: https://github.com/wavesplatform/matcher/blob/master/dex-it/src/test/scala/com/wavesplatform/it/sync/CancelOrderTestSuite.scala#L73  
* or decrease TTL: https://github.com/wavesplatform/matcher/blob/master/dex-it/src/test/scala/com/wavesplatform/it/sync/CancelOrderTestSuite.scala#L306

Suggest to decrease TTL.

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala [hunk 2]
```diff
@@ -1,33 +1,61 @@
...
 import com.wavesplatform.dex.domain.order.{Order, OrderType}
 import com.wavesplatform.it.MatcherSuiteBase
+import org.scalacheck.Gen
 
 class MatcherRestartTestSuite extends MatcherSuiteBase {
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
     broadcastAndAwait(IssueEthTx)
+    broadcastAndAwait(IssueWctTx)
```
Comment (vsuharnikov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:14]: Here we send a transaction and wait a block and do this for the second transaction.

Let's send two transaction simultaneously:
```scala
broadcastAndAwait(IssueEthTx, IssueWctTx)
```
and wait a block with them.

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala [hunk 3]
```diff
@@ -1,33 +1,61 @@
...
+      placeAndAwaitAtDex(aliceSecondOrder, HttpOrderStatus.Status.Filled)
+      eventually {
+        dex1.api.waitForOrderStatus(bobsOrderToFulfil, Status.Filled)
+      }
+
+      markup("place other Bob's orders")
+      bobsOrders.foreach(v => placeAndAwaitAtDex(v))
+
       // Check that order is correct
+      markup("check Alice order")
```
Comment (vsuharnikov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:45]: Let's use `print` from https://github.com/wavesplatform/matcher/blob/master/dex-it-common/src/main/scala/com/wavesplatform/dex/it/test/InformativeTestStart.scala#L29 instead. It writes a log to all files.

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala [hunk 4]
```diff
@@ -1,33 +1,61 @@
...
 
   "check order execution" - {
     "make order and after matcher's restart try to cancel it" in {
       // Alice places sell order
       val aliceOrder = mkOrder(alice, ethWavesPair, OrderType.SELL, 500, 2.waves * Order.PriceConstant)
+      val bobsOrderToCancel = mkBobOrder()
+      val bobsOrderToFulfil = mkBobOrder()
+      val bobsOrders = (1 to 10).map(_ => mkBobOrder())
+
+      markup("place Alice order")
```
Comment (alexander-branevskiy) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:25]: "place Alice's order"

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala (hunk 3 above)
Comment (alexander-branevskiy) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:45]: Alice's

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala [hunk 5]
```diff
@@ -50,6 +78,18 @@ class MatcherRestartTestSuite extends MatcherSuiteBase {
 
       dex1.api.waitForOrderStatus(aliceOrder, Status.Cancelled)
       dex1.api.getOrderHistoryByPublicKey(alice).head.status shouldBe Status.Accepted.name
+
+      markup("check Bob's orders")
+      bobsOrders.filter(order => order.id != bobsOrderToFulfil.id && order.id != bobsOrderToCancel.id).foreach { order =>
```
Comment (alexander-branevskiy) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:83]: we should check the total amount of expected orders here.

File: dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala (hunk 1 above)
Comment (alexander-branevskiy) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/MatcherRestartTestSuite.scala:93]: probably we can even don't use Gen here and stick to determining values.

**Review by alexander-branevskiy — CHANGES_REQUESTED** [review]

**Review by alexander-branevskiy — APPROVED** [review]

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/test/InformativeTestStart.scala [hunk 6]
```diff
@@ -26,7 +26,7 @@ trait InformativeTestStart extends TestSuite { self: BaseContainersKit =>
   }
 
   // Single quotes to emphasize the text in IDE
-  private def print(text: String): Unit = writeGlobalLog(s"'---------- [${LocalDateTime.now(ZoneId.of("UTC"))}] $text ----------'")
+  protected def print(text: String): Unit = writeGlobalLog(s"'---------- [${LocalDateTime.now(ZoneId.of("UTC"))}] $text ----------'")
```
Comment (alexander-branevskiy) [inline dex-it-common/src/main/scala/com/wavesplatform/dex/it/test/InformativeTestStart.scala:29]: discussed

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**